iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/NodeBreakerVoltageLevelAdapter.java

```diff
@@ -0,0 +1,407 @@
+            return getDelegate().getSwitchCount();
+        }
+
+        @Override
+        public void removeSwitch(String switchId) {
+            throw createNotSupportedNodeBreakerTopologyException();
+        }
+
+        @Override
+        public void removeAllSwitches() {
+            throw createNotSupportedNodeBreakerTopologyException();
+        }
+
+        @Override
+        public Bus getBus1(String switchId) {
+            return getBus(getDelegate().getBus1(switchId));
+        }
+
+        @Override
+        public Bus getBus2(String switchId) {
+            return getBus(getDelegate().getBus2(switchId));
+        }
+
+        @Override
+        public Switch getSwitch(String switchId) {
+            return getIndex().getSwitch(getDelegate().getSwitch(switchId));
+        }
+
+        @Override
+        public SwitchAdder newSwitch() {
+            throw createNotSupportedNodeBreakerTopologyException();
+        }
+
+        private PowsyblException createNotSupportedNodeBreakerTopologyException() {
+            return new PowsyblException("Not supported in a node/breaker topology");
+        }
+    }
+
+    static class BusViewAdapter extends BusCache<BusView> implements BusViewExt {
+
+        BusViewAdapter(BusView delegate, MergingViewIndex index) {
+            super(delegate, index);
+        }
+
+        @Override
+        public Iterable<Bus> getBuses() {
+            return Iterables.transform(getDelegate().getBuses(), this::getBus);
+        }
+
+        @Override
+        public Stream<Bus> getBusStream() {
+            return getDelegate().getBusStream().map(this::getBus);
+        }
+
+        @Override
+        public Bus getBus(String id) {
+            return getBus(getDelegate().getBus(id));
+        }
+
+        @Override
+        public Bus getMergedBus(String configuredBusId) {
+            return getBus(getDelegate().getMergedBus(configuredBusId));
+        }
+    }
+
+    private final NodeBreakerViewAdapter nodeBreakerView;
+
+    private final BusBreakerViewAdapter busBreakerView;
+
+    private final BusViewAdapter busView;
+
+    NodeBreakerVoltageLevelAdapter(final VoltageLevel delegate, final MergingViewIndex index) {
+        super(delegate, index);
+        nodeBreakerView = new NodeBreakerViewAdapter(delegate.getNodeBreakerView(), index);
+        busBreakerView = new BusBreakerViewAdapter(delegate.getBusBreakerView(), index);
+        busView = new BusViewAdapter(delegate.getBusView(), index);
+    }
+
+    @Override
+    public NodeBreakerView getNodeBreakerView() {
+        return nodeBreakerView;
+    }
+
+    @Override
+    public BusBreakerViewExt getBusBreakerView() {
+        return busBreakerView;
+    }
+
+    @Override
+    public BusViewExt getBusView() {
+        return busView;
+    }
+
+    @Override
+    public void invalidateCache() {
+        busBreakerView.invalidateCache();
+        busView.invalidateCache();
+        System.out.println("INVALIDATE");
```

Comment:
Same remark as before. Shouldn't a logger be used?

Reply:
Same answer, just for the debug